### PR TITLE
Format `f16` values correctly in intrinsic-test

### PR DESCRIPTION
**Pull request by sayantn**
Fix some x86 intrinsics too

 - `_mm_{u}comineq_sh` (these should be unordered, i.e. should return true if either operand is nan)
 - `_mm_mask_cvt{epi16_epi8, pd_ps, pd_epi32}` (top 64 bits should be 0)

### CODE REVIEW

**Comment by sayantn** [comment]
r? @folkertdev 

Just lmk before merging, will need to remove the last commit

**Comment by folkertdev** [comment]
cc @tgross35 was there some caveat to this approach?

**Comment by tgross35** [comment]
Not sure about the rest but just regarding formatting, these were added at a time that we didn't have `Debug` for f16

**Comment by folkertdev** [comment]
Right just double-checking that there is not some edge case where we'd get incorrect outputs by upcasting to `f32`.

File: crates/intrinsic-test/missing_x86.txt [hunk 1]
```diff
@@ -594,311 +23,45 @@ _rotwl
...
-_kshiftri_mask8
+# top bits are undefined, unclear how to test these
 _mm256_castsi128_si256
-_mm256_extract_epi16
-_mm256_extract_epi8
 _mm512_castsi128_si512
 _mm512_castsi256_si512
-# _mm512_conj_pch
+
+# Clang bug
```
Comment (folkertdev) [inline crates/intrinsic-test/missing_x86.txt:59]: Do we document what clang bug(s) these are?

Comment (sayantn) [reply crates/intrinsic-test/missing_x86.txt:59]: yeah I was thinking about putting some clang PRs to correct these. in summary,

 - `_mm{256}_extract_epi{8,16}` does `(int) (unsigned {char,short})`, but afaik C does sign-extension whenever the *target type* is signed, so this is getting sign-extended, whereas Intel specifies that it should zero-extend.
 - `_mm512_mask_reduce_{max,min}_{ps,pd}` uses `{f32,f64}::{MIN,MAX}` in place of masked-out values, but Intel specifies it should use `{f32,f64}::{NEG_INFINITY,INFINITY}`
